gserver/src/server.ts

```diff
@@ -83,13 +83,13 @@ interface Page {
 
 //Return start, end position and matched (if any) Gherkin step
 function handleLine(line: String): StepLine {
-    let typeRegEx = /Given |When |Then |And /;
+    let typeRegEx = /Given |When |Then |And |But /;
     let typeMatch = line.match(typeRegEx);
     let typePart = typeMatch[0];
     let stepPart = line.replace(gerkinRegEx, '');
     let stepMatch;
     for (let i = 0; i < steps.length; i++) {
-        if (stepPart.search(steps[i].reg) !== -1) {
+        if (line.trim().match(steps[i].reg) || stepPart.search(steps[i].reg) !== -1) {

```

Comment:
Got it, thank you. I see no correct way to fix all those items in current extension realisation. Looks like, that, to avoid dirty hacks, I should do some global changes. I'll start to do this in separate issue/branch.